v2/clients/http/deviceprofile.go

```diff
@@ -44,3 +49,87 @@ func (client *DeviceProfileClient) Update(ctx context.Context, reqs []requests.D
 	}
 	return responses, nil
 }
+
+func (client *DeviceProfileClient) DeviceProfileByName(ctx context.Context, name string) (res responses.DeviceProfileResponse, edgexError errors.EdgeX) {
+	u, err := url.Parse(client.baseUrl)
+	if err != nil {
+		return res, errors.NewCommonEdgeX(errors.KindClientError, "fail to parse baseUrl", err)
+	}
+	u.Path = path.Join(u.Path, v2.ApiDeviceProfileRoute, v2.Name, url.QueryEscape(name))
+	err = utils.GetRequest(ctx, &res, u.String())
+	if err != nil {
+		return res, errors.NewCommonEdgeXWrapper(err)
+	}
+	return res, nil
```

Comment:
Refactor into helper function like in other comment but without query params. Will be useful as out GETs are added.

Reply:
Refactored.